### ISSUE TITLE
Extend SPCGeonode setup to be customizable using Kubernetes

### ISSUE THREAD

**Issue by lucernae** [GeoNode/geonode#6208]
Hi, greetings everyone.

**Is your feature request related to a problem? Please describe.**
TL;DR:
I want to add more entrypoint scripts options/parameters in SPCGeoNode setup without breaking existing setup. I plan to use it for Kubernetes/Helm Charts.

_Long version_

My issue here is about small improvements for SPCGeoNode setup to expose more configuration to Kubernetes-based deployment.
In my opinion this not big enough to become GNIP, and also not a bug, so I'd rather put it as feature request proposals.

The SPCGeoNode setup is alot more stable now. I've decided to try to use the related docker images to create a Helm Charts for GeoNode kubernetes deployment. I've made a setup that works reasonably well with support from Kartoza for testing. 

The idea si fairly simple. Since we already have extensible images with environment variable options fairly exposed to the recipe (either docker-compose.yml or Kubernetes manifests yaml), I want to push some changes that I think might be useful upstream. The changes includes adding more options in the entrypoint script so it can receive this options via environment variable.
With this approach, both docker-compose.yml based setup of the original SPCGeoNode and also Kubernetes based setup can use the same options.
At the moment my Helm charts recipe is able to work because I modified the entrypoint scripts a little bit and store it as Kubernetes manifests. That way my Helm charts can still work with upstream `geonode/spcgeonode_*` images without having to rebuild the images. 
However it will be more beneficial for me to have these scripts pushed upstream, so I don't have to include the scripts in my manifests for future GeoNode releases.
It also might invite discussion in the community about the idea of using Kubernetes based setup, if some members want to give more feedback or ideas about this.

**Describe the solution you'd like**
My aim is to just add more options in the entrypoint script so Kubernetes can use it well with it's benefit, which is mainly abstracting the deployment and handling network requests and volumes in a cloud native way.
But the goal is also to not change existing functionality of SPCGeoNode deployment.

I will add more parameters and fill it with default value in order for the existing SPCGeoNode setup to not break even without using these parameters. I believe it shouldn't have any impact for current SPCGeoNode setup. If there are integration tests for this setup I want to test against those in order to not break existing setup.

The PR more or less will contain:
- Modified entrypoint scripts for GeoServer to handle OAuth2 settings and server certificate
- Modified initialization scripts for GeoNode to handle OAuth2 settings and abstraction for internal network between GeoNode and GeoServer

The PR will not contain:
- Kubernetes manifests files.

I will prepare the draft PR later in parallel with the discussion.

**Describe alternatives you've considered**
I think there is only one possible path here.

**Additional context**
For those interested: Kartoza's Helm Charts that contains the template to deploy GeoNode:
https://github.com/kartoza/charts/tree/master/charts/geonode/v0.1.1
